### PR TITLE
Rules [m-o]: replace old verifier

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**

### CODE REVIEW

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/MemberInitializedToDefaultTest.cs [hunk 1]
```diff
@@ -25,21 +25,26 @@ namespace SonarAnalyzer.UnitTest.Rules
...
+
 #endif
 
         [TestMethod]
         public void MemberInitializedToDefault_CodeFix() =>
-            OldVerifier.VerifyCodeFix<MemberInitializedToDefaultCodeFix>(
-                @"TestCases\MemberInitializedToDefault.cs",
-                @"TestCases\MemberInitializedToDefault.Fixed.cs",
-                new MemberInitializedToDefault());
+            builder
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/MemberInitializedToDefaultTest.cs:44]: In some cases we let the first invocation be on the first line, and at some places we  put all invocations to separate lines, from our coding style description it is not super clear to me what should be the way to go?

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/MemberShouldNotHaveConflictingTransparencyAttributesTest.cs [hunk 2]
```diff
@@ -25,18 +25,23 @@ namespace SonarAnalyzer.UnitTest.Rules
...
         public void MemberShouldNotHaveConflictingTransparencyAttributes_AssemblyLevel() =>
-            OldVerifier.VerifyNonConcurrentAnalyzer(@"TestCases\MemberShouldNotHaveConflictingTransparencyAttributes_AssemblyLevel.cs", new MemberShouldNotHaveConflictingTransparencyAttributes());
+            builder.AddPaths("MemberShouldNotHaveConflictingTransparencyAttributes_AssemblyLevel.cs").WithConcurrentAnalysis(false).Verify();
 
 #if NET
+
         [TestMethod]
         public void MemberShouldNotHaveConflictingTransparencyAttributes_CSharp10() =>
-            OldVerifier.VerifyAnalyzerFromCSharp10Library(@"TestCases\MemberShouldNotHaveConflictingTransparencyAttributes.CSharp10.cs", new MemberShouldNotHaveConflictingTransparencyAttributes());
+            builder.AddPaths("MemberShouldNotHaveConflictingTransparencyAttributes.CSharp10.cs").WithConcurrentAnalysis(false).WithOptions(ParseOptionsHelper.FromCSharp10).Verify();
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/MemberShouldNotHaveConflictingTransparencyAttributesTest.cs:42]: Do we only put invocations in separate lines if S103 is violated (that is what the coding style doc says)? My understanding as of now was that it was only a single line chain if there was nothing done except add path and verify.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/ObsoleteAttributesNeedExplanationTest.cs [hunk 3]
```diff
@@ -25,18 +25,20 @@ namespace SonarAnalyzer.UnitTest.Rules
...
         public void ObsoleteAttributesNeedExplanation_CSharp9() =>
-            OldVerifier.VerifyAnalyzerFromCSharp9Console(@"TestCases\ObsoleteAttributesNeedExplanation.CSharp9.cs", new ObsoleteAttributesNeedExplanation());
+            builder.AddPaths("ObsoleteAttributesNeedExplanation.CSharp9.cs").WithTopLevelStatements().Verify();
 
         [TestMethod]
         public void ObsoleteAttributesNeedExplanation_CSharp10() =>
-            OldVerifier.VerifyAnalyzerFromCSharp10Library(@"TestCases\ObsoleteAttributesNeedExplanation.CSharp10.cs", new ObsoleteAttributesNeedExplanation());
+            builder.AddPaths("ObsoleteAttributesNeedExplanation.CSharp10.cs").WithConcurrentAnalysis(false).WithOptions(ParseOptionsHelper.FromCSharp10).Verify();
 
 #endif
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/ObsoleteAttributesNeedExplanationTest.cs:43]: Empty line missing after directive.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/ObsoleteAttributesNeedExplanationTest.cs [hunk 4]
```diff
@@ -25,18 +25,20 @@ namespace SonarAnalyzer.UnitTest.Rules
...
     public class ObsoleteAttributesNeedExplanationTest
     {
+        private readonly VerifierBuilder builder = new VerifierBuilder<ObsoleteAttributesNeedExplanation>();
+
         [TestMethod]
         public void ObsoleteAttributesNeedExplanation() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\ObsoleteAttributesNeedExplanation.cs", new ObsoleteAttributesNeedExplanation());
+            builder.AddPaths("ObsoleteAttributesNeedExplanation.cs").Verify();
 
 #if NET
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/ObsoleteAttributesNeedExplanationTest.cs:34]: Empty line missing after directive.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/MemberInitializedToDefaultTest.cs (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/MemberInitializedToDefaultTest.cs:44]: Indeed. For me, the code is easier to read when all method invocations are aligned. We could discuss and make this explicit in our guidelines.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/MemberShouldNotHaveConflictingTransparencyAttributesTest.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/MemberShouldNotHaveConflictingTransparencyAttributesTest.cs:42]: I think we easily get used to reading this kind of test and having them on the same line should be acceptable. I'm also perfectly fine with a single invocation per line but sometimes I feel it doesn't help much.

**Pull request merged**